docs/operators/kinesisInputOperator.md

```diff
@@ -0,0 +1,178 @@
+
+#### Configuration properties
+
+-   ***accessKey*** - String[]
+    -   Mandatory Parameter.
+    -   Specifies the AWS credentials AccessKeyId.
+    
+-   ***secretKey*** - String[]
+    -   Mandatory Parameter.
+    -   Specifies the AWS credentials SecretAccessKey.
+
+-   ***streamName*** - String[]
+    -   Mandatory Parameter.
+    -   Specifies the name of the stream from where the records to be accessed.
+    
+-   ***strategy*** - PartitionStrategy
+    -   Operator supports two types of partitioning strategies, `ONE_TO_ONE` and `MANY_TO_ONE`.
+    
+        `ONE_TO_ONE`: If this is enabled, the AppMaster creates one input operator instance per kinesis shard partition. So the number of shard partitions equals the number of operator instances.
+        `MANY_TO_ONE`: The AppMaster creates K = min(initialPartitionCount, N) Kinesis input operator instances where N is the number of shard partitions. If K is less than N, the remaining shard partitions are assigned to the K operator instances in round-robin fashion. If K is less than initialPartitionCount, the AppMaster creates one input operator instance per shard partition. For example, if initialPartitionCount = 5 and number of shard partitions(N) = 2 then AppMaster creates 2 Kinesis input operator instances.
+        Default Value = `PartitionStrategy.ONE_TO_ONE`.
+
+-   ***initialPartitionCount*** - Integer
+    -   When the MANY_TO_ONE partition strategy is enabled, this value indicates the number of Kinesis input operator instances. 
+        Default Value = 1.
+
+-   ***initialOffset*** - InitialOffset
+    -   Indicates the type of offset i.e, `EARLIEST` or `LATEST`. 
+        `LATEST` => Consume new record(s) from latest offset in the shard. 
+        `EARLIEST` => Consume all record(s) available in the shard.
+        Default value = `InitialOffset.LATEST`
+        
+-   ***repartitionInterval*** - Long
+    -   Interval specified in milliseconds. This value specifies the minimum time required between two repartition actions. 
+        Default Value = 30 Seconds.
+
+-   ***repartitionCheckInterval*** - Long
+    -   Interval specified in milliseconds. This value specifies the minimum interval between two stat checks.
+        Default Value = 5 Seconds.
+
+-   ***maxTuplesPerWindow*** - Integer
+    -   Controls the maximum number of records emitted in each streaming window from this operator. Minimum value is 1. 
+        Default value = `MAX_VALUE`
+
+-   ***holdingBufferSize*** - Long
+    -   Indicates the maximum number of messages kept in memory for emitting.
+        Default value = 1024.
+
+-   ***windowDataManager*** - WindowDataManager
+    -   If set to a value other than the default, such as `FSWindowDataManager`, specifies that the operator will process the same set of messages in a window before and after a failure. This is important but it comes with higher cost because at the end of each window the operator needs to persist some state with respect to that window.
```

Comment:
This could be rephrased. The description should convey following .
1. What is the purpose of this property ?
2. How does the default work ?
3. What are my choices if i want to specify any other windowDataManager and how each of those choices will affect the behaviour?